### DISCUSSION TITLE
Add a way to disable any node

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#4722] [Engine Core]
### Problems:

1. When I create scenes I often experiment with nodes properties values (position, size etc).
So sometimes I just duplicate the node, set the old node visibility to false and then change some parameters for the new node.
Then I compare results, and delete the old node if the new one looks/works better.
But there are many cases when I can't easily use this approach, for example if there is a reference to the node by name  like `onready var node =$path_to_the_node`, or if the node uses signals.
I want the node to still be in the tree in the editor, but completely disabled, which means it will not exist in the game at all.

2. Sometimes I just want to quickly test how the game would behave if I remove some node from the editor.
I solve it deleting the node then run the game then when I close the game window I press Ctrl+z to undo the node deletion, but this is not handy.

### Solution:
The idea is to have an option, for example a checkbox (in the node right-click context menu), pressing it the node would still be in the Scene Tree Editor, but completely disabled like it doesn't exists as a node, (including the tool mode I guess?).
That means $Node won't work and `get_children()` will not have disabled nodes in the children list.

- Name collisions in the disabled nodes should be allowed: the user can have multiple disabled nodes with name Node and at the same time one enabled node with name Node.
- There should be an easy way to clearly see in the editor  that a node is disabled.
- All changed properties of disabled node are still stored in the scene, the node simply not created in the game, even Node.new() is not called.
- I don't think it should be as a property, I don't see useful cases,  this should be kinda editor/Packed Scene feature only
- See also set node as a Placeholder, this is kinda similar, but it works only for scenes and Placeholder's goal is different (delayed creation of the nodes)

**Comment by Calinou** [comment]
The use case seems a bit contrived to me. I don't see why you'd need to do this after the [process mode refactor](https://github.com/godotengine/godot/pull/46191) has landed in 4.0.alpha.

**Reply by me2beats** [reply]
- the node still exists in the game and in the editor
- the node can be accessed from the code of other nodes 
- _init(), enter_tree() and _ready() are called
- unable to create a node named Node, if I already have a node with disabled proccess named Node (I think it could be allowed for disabled nodes)
- signals are still emitted

**Reply by me2beats** [reply]
the idea is to not create the node in the game at all

**Comment by wareya** [comment]
As a workaround, could you add code to instantly destroy undesired nodes in _init()? (have not tested if this would actually work)

> But there are many cases when I can't easily use this approach, for example if there is a reference to the node by name like `onready var node =$path_to_the_node`

If the node doesn't exist in the game *at all*, then this will break regardless. The only way for this to not break is if the node *does* exist, but spawns in a 'paused' state of some kind.

**Reply by me2beats** [reply]
sometimes the idea is to let it break, when testing. In the release version the code should be fixed of course
